src/http.facility.js

```diff
@@ -43,9 +44,18 @@ class HttpFacility extends Base {
         opts = {}
       }
 
-      const url = path.includes('://') ? path : `${this.baseUrl}/${path.replace(/^\//, '')}`
+      let url = path.includes('://') ? path : `${this.baseUrl}/${path.replace(/^\//, '')}`
+
+      const urlHasQuestion = url.includes('?')
+      if (this.qs) url += (urlHasQuestion ? '' : '?') + this.qs
+
+      const reqOpts = _.pick(opts, ['body', 'headers', 'method', 'redirect', 'agent', 'compress', 'timeout', 'qs'])
+      if (reqOpts.qs) {
+        if (!urlHasQuestion) url += '?'
+        if (this.qs) url += '&'
```

Comment:
hmm, we should've added `qs` to the url above so maybe we could check here and if there is `qs`, add `&` and if there is no `qs` add `?`?

as i can see there are several branches to consider:
* we got clean URL with no query params
  * qs is null / empty - we do nothing
  * qs is non-empty - we add `?` and chain query params
* we got URL that has query params because someone appended them
  * qs is null / empty - we do nothing
  * qs is non-empty - we add `&` and chain

after that we know if the URL has query params added or not. If it has - we chain starting with `&`. If not - with `?`

so I'd suggest simplifying it a bit:
```
let urlHasQParams = url.includes('?')
if (this.qs) {
  url += urlHasQParams ? '&' : '?' + this.qs
  urlHasQParams = true
}
if (reqOpts.qs) {
  url += urlHasQParams ? '&' : '?' + new URLSearchParams(reqOpts.qs).toString()
}
```

Reply:
I've adjusted the flow to be even simpler and added support for overriding this.qs params by reqOpts.qs